R/plumber-step.R

```diff
@@ -307,6 +315,68 @@ PlumberEndpoint <- R6Class(
       self$path <- path
       private$regex <- createPathRegex(path, self$getFuncParams())
       path
+    },
+    #' @description convert args to specific types
+    #' @param args List of args to convert
+    convertArgs = function(args) {
+      if (!isTRUE(getOption("plumber.typedParameters", TRUE))) {
+        return(args)
+      }
+
+      if (is.null(args)) {
+        # no args to convert
+        return(args)
+      }
+
+      if (length(args) == 0) {
+        # no args to convert
+        return(args)
+      }
+
+      args_names <- names(args)
+      if (is.null(args_names)) {
+        # no names - won't be able to dp any conversions
+        return(args)
+      }
+
+      params <- self$getEndpointParams()
+
+      if (length(params) == 0) {
+        # no information to convert by
+        return(args)
+      }
+
+      # TODO - I suspect someone with more base-R could turn this
+      # for into some wonderful s/lapply thing...
+      converted_args <- list()
+
+      for (i in 1:length(args)) {
+        key <- args_names[[i]]
+        arg <- args[[i]]
+        if (key == "") {
+          # not sure why... but we allow unnamed arguments through
+          converted_args <- c(converted_args, arg)
+          next;
+        }
+
+        param <- params[[key]]
+        converter <- identity
+        if (!is.null(param)) {
+          # NOTE: areArrays is FALSE here as we do not want to parse comma
+          #    separated strings (or do we?)
```

Comment:
Agreed - cross reference with https://github.com/rstudio/plumber/blob/76cecc58925897c91cb88203ab485a0c512a0d26/R/openapi-types.R#L31 - what scenario is `converterArray` for?

Reply:
I think I'd like to get rid of this - I think it can only be for supporting arrays in dynamic routes - since queryparams do arrays using repeated key value pairs and JSON bodies do arrays using Jason (not sure about forms though?)